File: src/main/java/no/nav/nav/sbl/dialogarena/modiabrukerdialog/service/SakerService.java
package no.nav.nav.sbl.dialogarena.modiabrukerdialog.service;

import no.nav.nav.sbl.dialogarena.modiabrukerdialog.domain.Sak;
import no.nav.nav.sbl.dialogarena.modiabrukerdialog.domain.Saker;

import java.util.List;

public interface SakerService {
    public Saker hentSaker(String fnr);
    public List<Sak> hentListeAvSaker(String fnr);
}

File: web/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/api/service/saker/SakerService.kt
package no.nav.sbl.dialogarena.modiabrukerdialog.api.service.saker

import no.nav.sbl.dialogarena.modiabrukerdialog.api.domain.saker.Sak
import no.nav.sbl.dialogarena.modiabrukerdialog.api.exceptions.JournalforingFeilet

interface SakerService {
    class Resultat(val saker: MutableList<Sak> = mutableListOf(), val feiledeSystemer: MutableList<String> = mutableListOf())

    fun hentSaker(fnr: String): Resultat

    fun hentSakSaker(fnr: String): Resultat

    @Throws(JournalforingFeilet::class)
    fun knyttBehandlingskjedeTilSak(fnr: String?, behandlingskjede: String?, sak: Sak, enhet: String?)
}